src/main/perl/RuleBasedEditor.pm

```diff
@@ -0,0 +1,1215 @@
+                # remove_if_undef to be set.
+                # Note that this will never match instance parameters and will not remove entries
+                # no longer part of the configuration in a still existing LINE_VALUE_ARRAY or
+                # LINE_VALUE_STRING_HASH.
+                unless ($attribute_present) {
+                    if ($rule_parsing_options->{remove_if_undef}) {
+                        *$self->{LOG}->debug(1,
+                                                 "$function_name: attribute '"
+                                               . $rule_info->{attribute}
+                                               . "' undefined, removing configuration line");
+                        $self->_commentConfigLine($keyword, $line_fmt);
+                    }
+                    next;
+                }
+
+                # Instance parameters are specific, as this is a nlist of instance
+                # with the value being a nlist of parameters for the instance.
+                # Also the variable name must be updated to contain the instance name.
+                # One configuration line must be written/updated for each instance.
+                if ($value_fmt == LINE_VALUE_INSTANCE_PARAMS) {
+                    foreach my $instance (sort keys %{$attr_value}) {
+                        my $params = $attr_value->{$instance};
+                        *$self->{LOG}->debug(1, "$function_name: formatting instance '$instance' parameters ($params)");
+                        $config_value =
+                          $self->_formatAttributeValue(
+                                                       $params,
+                                                       $line_fmt,
+                                                       $value_fmt,
+                                                       $value_opt,
+                                                      );
+                        my $config_param = $keyword;
+                        my $instance_uc  = uc($instance);
+                        $config_param =~ s/%%INSTANCE%%/$instance_uc/;
+                        *$self->{LOG}->debug(2, "New variable name generated: >>>$config_param<<<");
+                        $self->_updateConfigLine($config_param, $config_value, $line_fmt);
+                    }
+                    $config_updated = 1;
+                } elsif ($value_fmt == LINE_VALUE_STRING_HASH) {
+                    # With this value format, several lines with the same keyword are generated,
+                    # one for each key/value pair.
+                    foreach my $k (sort keys %$attr_value) {
+                        my $v = $attr_value->{$k};
+                        # Value is made by joining key and value as a string
+                        # Keys may be escaped if they contain characters like '/': unescaping a non-escaped
+                        # string is generally harmless.
```

Comment:
"generally harmless" is nice one. this is not good at all, esp given the current crappy escape/unescape code. please document this in the pod.